include/proxy-wasm/wasm.h

```diff
@@ -289,6 +291,9 @@ inline const std::string &WasmBase::vm_configuration() const {
 }
 
 inline void *WasmBase::allocMemory(uint64_t size, uint64_t *address) {
+  if (!malloc_) {
+    return nullptr;
+  }
```

Comment:
Do we need this? We fail hard if `malloc` is not exported.

(I'm not saying it's not good to add it just to be safe, I'm just curious why did you add it now)

Reply:
Generally I was trying to make the proxy-independent code resilient against unexpected behavior.  An integrator could choose to not hard fail on error().  For envoy we throw an exception which we will catch at a higher level, but presumably someone code set a flag.